src/main/java/io/github/thebusybiscuit/slimefun4/api/accessibility/AccessManager.java

```diff
@@ -0,0 +1,166 @@
+     *
+     * @param player The instance of the player.
+     * @return Returns the access level this player has.
+     */
+    @Nonnull
+    AccessLevel getAccessLevel(@Nonnull Player player);
+
+    /**
+     * Get the {@link AccessLevel} for a player, defaulting to {@link ConcreteAccessLevel#NONE} if {@link #hasDataFor(Object)} is false.
+     *
+     * @param player The unique id of the player
+     * @return Returns the access level this player has.
+     */
+    @Nonnull
+    AccessLevel getAccessLevel(@Nonnull UUID player);
+
+    /**
+     * Get the {@link AccessData} object backing a certain class. Mutations to the returned
+     * object will be reflected by this manager.
+     *
+     * @param clazz The instance of the class
+     * @param <T>   Generic type of the class
+     * @return Returns a {@link Optional} of the AccessData object backing the provided class.
+     * @see #getOrRegisterAccessData(Class, AccessData)
+     */
+    @Nonnull
+    <T> Optional<AccessData<T>> getAccessData(@Nonnull Class<T> clazz);
+
+    /**
+     * Get the {@link AccessData} object backing a certain class, or register a default if
+     * {@link #getAccessData(Class)} is not present. Mutations to the returned
+     * object will be reflected by this manager.
+     *
+     * @param clazz The instance of the class
+     * @param <T>   Generic type of the class
+     * @return Returns the AccessData object backing the provided class
+     */
+    @Nonnull
+    <T> AccessData<T> getOrRegisterAccessData(@Nonnull Class<T> clazz,
+                                              @Nonnull final AccessData<T> def);
+
+    /**
+     * Check whether a {@link Player} has a given {@link AccessLevel} or higher.
+     *
+     * @param player      The instance of the player
+     * @param accessLevel The instance of the AccessLevel
+     * @return Returns true if the player has the provided access level or higher.
+     */
+    default boolean hasLevel(@Nonnull Player player, @Nonnull AccessLevel accessLevel) {
+        return getAccessLevel(player).compare(accessLevel) >= 0;
+    }
+
+    /**
+     * Check whether a player has a given {@link AccessLevel} or higher.
+     *
+     * @param player      The unique id of the player
+     * @param accessLevel The instance of the AccessLevel
+     * @return Returns true if the player has the provided access level or higher.
+     */
+    default boolean hasLevel(@Nonnull UUID player, @Nonnull AccessLevel accessLevel) {
+        return getAccessLevel(player).compare(accessLevel) >= 0;
+    }
+
+    /**
+     * Check whether the manager has data for a given object. This method is useful in
+     * understanding whether a player is on the access list when {@link #getAccessLevel(Player)}
+     * provides the default result.
+     *
+     * @param object The instance of the object.
+     * @return Returns whether the manager has data for a given object.
+     */
+    boolean hasDataFor(@Nonnull Object object);
```

Comment:
Can you use a generic type T here?
Even if it makes no difference here, this would be consisztent with the other methods and would allow us to fine-tune values later if we want that.